### PR TITLE
[PR port] Port #49737 - Isbm pkg resources crash

### PR DESCRIPTION
**Pull request by roaldnefs**
PR port #49737
- [ ] Docs
- [ ] Changelog
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: salt/states/pip_state.py [hunk 1]
```diff
@@ -144,11 +150,7 @@ def __virtual__():
     """
     Only load if the pip module is available in __salt__
     """
-    if HAS_PKG_RESOURCES is False:
-        return False, "The pkg_resources python library is not installed"
-    if "pip.list" in __salt__:
-        return __virtualname__
-    return False
+    return "pip.list" in __salt__ and __virtualname__ or False
```
Comment (twangboy) [inline salt/states/pip_state.py:153]: We prefer explicit failures here... something like this:
```
if not HAS_PKG_RESOURCES:
    return False, "The pkg_resources python library is not installed"
if not "pip.list" in __salt__:
    return False, "pip.list is not available"
return __virtualname__
```
Only return the virtualname if all conditions are met. When conditions aren't met, return False, and why it was not met.

**Review by twangboy — CHANGES_REQUESTED** [review]
Just one change requested.

File: salt/states/pip_state.py (hunk 1 above)
Comment (roaldnefs) [reply salt/states/pip_state.py]: Thanks @twangboy for the feedback! Fixed in 328b457.

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**